service/tasks/sync_project.py

```diff
@@ -0,0 +1,199 @@
+    #    leaders can view images (but not providers)
+    try:
+        lead = Leadership.objects.get(user=atm_user, group=group)
+    except:
+        lead = Leadership(userid_id=atm_user.id, group_id=group.group_ptr_id)
+        lead.save()
+    # ensure that the user id is added as a member of that group
+    group.user_set.add(atm_user)
+    group.save()
+    return group
+
+
+def get_atm_project(os_project, group_id):
+    """
+    This gets the project from the database
+    """
+    celery_logger.info("os_project: ")
+    celery_logger.info(repr(os_project))
+    try:
+        atm_project = Project.objects.get(os_project_id=os_project.id)
+    except:
+        # if we cannot get the project (group is a FK in the project)
+        atm_project = Project()
+        atm_project.owner_id = group_id
+    atm_project.name = os_project.name
+    atm_project.os_domain_id = os_project.domain_id
+    atm_project.os_project_id = os_project.id
+    atm_project.save()
+    celery_logger.info("atm_project: ")
+    celery_logger.info(repr(atm_project))
+    return atm_project
+
+
+def get_os_scoped_token(auth_url, unscoped_token, os_project_id):
+    scoped_token_auth = v3.Token(auth_url=auth_url,
+                                 token=unscoped_token,
+                                 project_id=os_project_id,
+                                 unscoped=False)
+    scoped_sess = session.Session(auth=scoped_token_auth)
+    scoped_token = scoped_sess.get_token()
+    return scoped_token
+
+
+def get_os_project_catalog(auth_url, scoped_token):
+    """
+    gets the openstack service catalog
+    """
+    response = requests.get(auth_url + '/auth/tokens',
+                            headers={'x-auth-token': scoped_token,
+                                     'x-subject-token': scoped_token})
+    return response.text
+
+
+def get_compute_url(json_catalog):
+    celery_logger.info("start of get_compute_url")
+    try:
+        catalog = json.loads(json_catalog)
+    except KeyError:
+        celery_logger.info("Invalid token passed")
+        raise serializers.ValidationError("Invalid token passed")
+    endpoint_catalog = catalog['token']['catalog']
+    compute = None
+    for item in endpoint_catalog:
+        if item['type'] == 'compute':
+            compute = item
+    if not compute:
+        celery_logger.info("Cannot find compute endpoint catalog")
+        raise serializers.ValidationError("Cannot find compute endpoint catalog")
+    compute_url = None
+    for ep in compute['endpoints']:
+        if ep['interface'] == 'public':
+            compute_url = ep['url']
+    if not compute_url:
+        celery_logger("Cannot find a public compute enpoint url")
+        raise serializers.ValidationError("Cannot find a public compute endpoint url")
+    celery_logger.info("end of get_compute_url")
+    return compute_url
+
+
+# RBB disable celery so that tests will pass
+# @task(name="sync_atm_with_openstack")
```

Comment:
Add a comment for future reference with # TODO. Pending performance comparison with and without Celery.